lightning/src/ln/channelmanager.rs

```diff
@@ -522,32 +522,50 @@ impl BestBlock {
 	pub fn height(&self) -> u32 { self.height }
 }
 
+#[derive(Copy, Clone, PartialEq)]
+enum NotifyOption {
+	DoPersist,
+	SkipPersist,
+}
+
 /// Whenever we release the `ChannelManager`'s `total_consistency_lock`, from read mode, it is
 /// desirable to notify any listeners on `await_persistable_update_timeout`/
-/// `await_persistable_update` that new updates are available for persistence. Therefore, this
+/// `await_persistable_update` when new updates are available for persistence. Therefore, this
 /// struct is responsible for locking the total consistency lock and, upon going out of scope,
 /// sending the aforementioned notification (since the lock being released indicates that the
 /// updates are ready for persistence).
-struct PersistenceNotifierGuard<'a> {
+///
+/// We allow callers to either always notify by constructing with `notify_on_drop` or chose to
+/// notify or not based on whether relevant changes have been made, providing a closure to
+/// `optionally_notify` which returns a `NotifyOption`.
+struct PersistenceNotifierGuard<'a, F: Fn() -> NotifyOption> {
 	persistence_notifier: &'a PersistenceNotifier,
+	should_persist: F,
 	// We hold onto this result so the lock doesn't get released immediately.
 	_read_guard: RwLockReadGuard<'a, ()>,
 }
 
-impl<'a> PersistenceNotifierGuard<'a> {
-	fn new(lock: &'a RwLock<()>, notifier: &'a PersistenceNotifier) -> Self {
+impl<'a> PersistenceNotifierGuard<'a, fn() -> NotifyOption> { // We don't care what the concrete F is here, its unused
```

Comment:
nit: it's
Also a bit confused by this comment because it _is_ used [here](https://github.com/rust-bitcoin/rust-lightning/pull/916/commits/e2328d67d79a09b140f159e189f2f30e1287dee0#diff-645d12c7269d09caab57d2e0c1fd34f672ee259994c86ad4bad7f4da9183671cR567), isn't it?

Reply:
No, the concrete type here is a function pointer (ie `fn() -> NotifyOption`), but we don't actually ever use a `PersistenceNotifierGuard` with a function pointer, only closures.